test/quantization/pt2e/test_quantize_pt2e_qat.py

```diff
@@ -720,7 +720,88 @@ def test_mixing_qat_ptq(self):
             # 3 x linear: 1 for act, 1 for output
             ns.call_function(
                 torch.ops.quantized_decomposed.quantize_per_tensor.default
+            ): 8,
+            ns.call_function(
+                torch.ops.quantized_decomposed.dequantize_per_tensor.default
             ): 9,
+            ns.call_function(
+                torch.ops.quantized_decomposed.dequantize_per_channel.default
+            ): 3,
+            # There needs to be one for hardtanh
+        }
+        self.checkGraphModuleNodes(
+            exported_model.graph_module, expected_node_occurrence=node_occurrence
+        )
+
+    def test_mixing_qat_ptq_with_checkpointing(self):
+        import io
+
+        buffer = io.BytesIO()
+        example_inputs = (torch.randn(2, 3, 4, 4),)
+        model = TestQuantizeMixQATAndPTQ.QATPTQTestModule()
+        model.train()
+
+        self._prepare_qat_linears(model)
+
+        loss = torch.nn.MSELoss()
+        optimizer = torch.optim.SGD(model.parameters(), lr=0.001, momentum=0.9)
+        for i in range(2):
+            after_prepare_result_pt2e = model(*example_inputs)
+            target = torch.rand_like(after_prepare_result_pt2e)
+            output = loss(after_prepare_result_pt2e, target)
+            output.backward()
+            optimizer.step()
+
+        state_dict_before = copy.deepcopy(model.state_dict())
```

Comment:
nit: could you also add some comments to specify the context for each section of code? like QAT, checkpoint, load checkpoint, PTQ etc.